### ISSUE TITLE
Work out what we're doing

### ISSUE THREAD

**Issue by MMK21Hub** [SheepCommander/KnowledgeBase#2]
This is a central issue for discussing how we are going to handle copying over pastes to GitHub.

## So far...
In the Discord server, `#dp-faqs`, `#rp-faqs` and `#other-faqs` need to be merged into one `#faqs` channel. The pastes will need a Raw Link and Watermark added. There will also be example packs available in this repo. The pastes should link those where relevant.

## Points for discussion
 - ~~Are we going to use a folder in this repo or GitHub Gist for the raw pastes?~~ _This repo_
 - ~~What will be the folder layout of the repo?~~ *See https://github.com/SheepCommander/KnowledgeBase/issues/2#issuecomment-720672008*
 - ~~FAQ or FAQs?~~ _FAQ_
 - ~~Are we going to also add GitHub-friendly versions of the pastes?~~ _Yes_
 - Licence? (See <https://docs.github.com/en/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/licensing-a-repository>)

## Other info
Current tasks:
|#|Task|Status|Issue
|-|-|-|-|
|:one:|Add raw pastes to GH|In progress|#5
|:three:|Add watermark to all raw pastes|Blocked by :one:|
|:four:|Add raw link to all raw pastes|Blocked by :one:|
|:five:|Create example packs and link them where appropriate|In progress|
|:six:|Move pastes to `#resources` and `#faqs` in the Discord|Blocked by :three:, :four: and :five:|

I've created a `contributing-draft` branch. Once all this is agreed on we can add it to a `Contributing.md`.

Feel free to comment here if you can help with anything.

**Comment by MMK21Hub** [comment]
- It makes sense for us to copy over the pastes to Github first, then we can collab to work on :three:, :four: and then :five:. Once the pastes are up to the new standard, we can copy them over to the new Discord channels (task :five:).

**Comment by Pepe20129** [comment]
The license should be MIT or GNU GPL.
(https://en.wikipedia.org/wiki/MIT_License)
(https://en.wikipedia.org/wiki/GNU_General_Public_License)

**Comment by Pepe20129** [comment]
Folder Layout:

Resources
  FAQ
    Resource Packs
    Data Packs
    Other
  Templates
    Resource Packs
    Data Packs
    Commands Explained
    Other

**Comment by MMK21Hub** [comment]
> Folder Layout:
> 
> FAQ
> 
> > Resource Packs
> > Data Packs
> > Other

What would go under "other"? 🤔

**Comment by Pepe20129** [comment]
#other-faqs in the discord

**Comment by SheepCommander** [comment]
We're going to use this Repo for raw pastes, with gists we can't collaborate

**Comment by Pepe20129** [comment]
I know.

**Comment by SheepCommander** [comment]
`#other-faqs` etc. aren't meant to stay. 

I suppose here's what I'd suggest:
> 📁`Resources`
> > 📁`FAQ`
> > > 📁`insert faq`
> > > > 📁`Raw Paste`
> > > > 📁`Downloadable Tempate`

**Comment by Anthony2be** [comment]
maybe add like examples and how they work?

**Comment by SheepCommander** [comment]
Yeah maybe

**Comment by MMK21Hub** [comment]
I think MIT would be better. The main difference is that with GNU, anything that contains part of it has to also be released under GNU. You can reuse MIT content under any licence. (If I understand correctly.)
![https://tldrlegal.com/license/mit-license](https://user-images.githubusercontent.com/50421330/97540882-2b8ae100-19bc-11eb-8e42-0a21e3840c9e.png)

**Comment by MMK21Hub** [comment]
> I suppose here's what I'd suggest:
> 
> > 📁`FAQ`
> > > 📁`insert faq`
> > > > 📁`Raw Paste`
> > > > 📁`Downloadable Tempate?`
> 
> > 📁`Templates`
> > > 📁`insert template`
> > > > 📁`Raw Paste`
> > > > 📁`Downloadable Tempate`

Wait now I'm confused. Here's what I was thinking of:
```md
KnowledgeBase
> 📁 Resources
> > 📁 FAQ
> > > 📁 Datapacks
> > > > 🔽 [readable faq].md
> > > 📁 Resource Packs
> > > > 🔽 [readable faq].md
> > 📁 Templates
> > > 📁 Datapacks
> > > > 🔽 [readable template].md
> > > 📁 Resource Packs
> > > > 🔽 [readable template].md
> 📁 Raw Pastes
> > 📁 FAQ
> > > 📁 Datapacks
> > > > 📄 [raw faq].md
> > > 📁 Resource Packs
> > > > 📄 [raw faq].md
> > 📁 Templates
> > > 📁 Datapacks
> > > > 📄 [raw template].md
> > > 📁 Resource Packs
> > > > 📄 [raw template].md
> 📁 Downloadable Templates
> > 📁 Datapacks
> > > 🤐 [pre-made datapack].zip
> > 📁 Resource Packs
> > > 🤐 [pre-made resource pack].zip
```

As you can see, that's a lot of repetition. So are you suggesting something like this?
```md
KnowledgeBase
> 📁 Resources
> > 📁 FAQ
> > > 📁 Datapacks
> > > > 🔽 [readable faq].md
> > > > 📄 [raw template].txt
> > > 📁 Resource Packs
> > > > 🔽 [readable faq].md
> > > > 📄 [raw template].txt
> > 📁 Templates
> > > 📁 Datapacks
> > > > 🔽 [readable template].md
> > > > 📄 [raw template].txt
> > > 📁 Resource Packs
> > > > 🔽 [readable template].md
> > > > 📄 [raw template].txt
> 📁 Downloadable Templates
> > 📁 Datapacks
> > > 🤐 [pre-made datapack].zip
> > 📁 Resource Packs
> > > 🤐 [pre-made resource pack].zip
```
In which case I agree. Or am I getting the wrong idea?

**Comment by Zefyro** [comment]
I think the last one is great.

**Comment by MMK21Hub** [comment]
<big>Big change!</big>
GitHub is just going to be used for raw pastes and downloadable templates: **not** the "readable" GFM versions. People should go to the Discord to browse the pastes.

@SheepCommander could you confirm? I'll update the task desc.

**Comment by SheepCommander** [comment]
Yes, it's only for Raw Pastes and Downloadable Templates (Maybe also examples?) readable will be on the Discord Server

**Comment by MMK21Hub** [comment]
New layout proposal then:

```
KnowledgeBase
> 📁 Raw Pastes
> > 📁 FAQ
> > > 📁 Datapacks
> > > > 📄 [raw FAQ].txt
> > > 📁 Resource Packs
> > > > 📄 [raw FAQ].txt
> > 📁 Templates
> > > 📁 Datapacks
> > > > 📄 [raw template].txt
> > > 📁 Resource Packs
> > > > 📄 [raw template].txt
> 📁 Downloadable Templates
> > 📁 Datapacks
> > > 🤐 [pre-made datapack].zip
> > 📁 Resource Packs
> > > 🤐 [pre-made resource pack].zip

**Comment by SheepCommander** [comment]
Well we could sort it by category like that, or sort it by the FAQ.

> 📁 Raw Pastes
> > 📁 FAQ
> > > 📁 Datapacks
> > > > 📁 [FAQ]
> > > > > 📄 [raw FAQ].txt
> > > > > 📁 Downloadable Template
> > > > > 🤐 Example Pack
> > > 📁 Resource Packs
> > > > 📁 [FAQ]
> > > > > 📄 [raw FAQ].txt
> > > > > 📁 Downloadable Template
> > > > > 🤐 Example Pack

**Comment by SheepCommander** [comment]
No, I think it should be sorted more like this:

> 📁 FAQ
> > 📁 [FAQ]
> > > 📄 [raw FAQ].txt
> > > 📁 Downloadable Template?
> > > 🤐 Example Pack

I'm not sure if I should just have 1 FAQ channel or FAQ and Templates since the difference between FAQ & Template is kinda confusing, and also should it be "How to setup Scoreboard ID system" or "How to add Multiplayer support", things like that are kinda tricky.

**Comment by MMK21Hub** [comment]
LGTM. I'll put the full tree here then:
```
KnowledgeBase
> 📁 Raw Pastes
> > 📁 FAQ
> > > 📁 Datapacks
> > > > 📄 [raw FAQ].txt
> > > > 🤐 [relevent template].zip
> > > > 🤐 [example pack].zip
> > > 📁 Resource Packs
> > > > 📄 [raw FAQ].txt
> > > > 🤐 [relevent template].zip
> > > > 🤐 [example pack].zip
```
Much cleaner! Two more questions:

 1. Should we scrap the `Raw Pastes` folder, now that it only has one child? (I.e. `📁FAQ` would go in the root.
 2. What's the difference between "downloadable template" and "example pack"?

**Comment by SheepCommander** [comment]
Yeah don't know lol, let's remove Example Pack and keep Downloadable Template.

**Comment by MMK21Hub** [comment]
Final folder structure:

> ​:octocat: KnowledgeBase
> > 📁 FAQ
> > > 📁 [Paste]
> > > > 📄 [raw FAQ].txt
> > > > 🤐 [example pack].zip
> > > 📁 [Another paste]
> > > > 📄 [raw FAQ].txt
> > > > 🤐 [relevent template].zip

**Comment by SheepCommander** [comment]
Yep, now we _can_ work on this for the most part without bottle neck. Main thing will be allowing people to use the chatbox in the channels to make it easier to copy and paste

**Comment by SheepCommander** [comment]
Also it's no longer #resources it's one #faq

**Comment by SheepCommander** [comment]
Naming Convention is being decided by a vote, but how would a contributor determine what to name the resource?

**Comment by SheepCommander** [comment]
I did Lock instead of Close, but anyways this is mostly outdated I'm going to replace it with a new issue on what license to choose
